File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/D_Blue_and_Red_Balls.cpp
#include <bits/stdc++.h>
#define MOD 1'000'000'007
#define MAX_N 2000
using namespace std;
typedef long long ll;

ll N, K;
ll P[MAX_N + 1], Q[MAX_N + 1];

ll mod_pow(ll x, ll n) {
  ll res = 1;
  while (n > 0) {
    if (n & 1) { res = res * x % MOD; }
    x = x * x % MOD;
    n >>= 1;
  }
  return res;
}

ll inv(ll x) {
  return mod_pow(x, MOD - 2);
}

void init() {
  P[0] = 1;
  for (ll i = 1; i <= N; ++i) {
    P[i] = P[i - 1] * i % MOD;
  }
  Q[N] = inv(P[N]);
  for (ll i = N; i >= 1; --i) {
    Q[i - 1] = Q[i] * i % MOD;
  }
}

ll mod_com(ll n, ll k) {
  if (k == 0) { return 1; }
  if (k < 0 || k > n) { return 0; }
  return P[n] * Q[n - k] % MOD * Q[k] % MOD;
}

void solve() {
  init();
  for (ll i = 1; i <= K; ++i) {
    ll res = mod_com(N - K + 1, i) * mod_com(K - 1, i - 1) % MOD;
    printf("%lld\n", res);
  }
}

int main() {
  cin >> N >> K;
  solve();
}

File: AtCoder/ACP/BootCamp4b/Hard100_2/python/D_Blue_and_Red_Balls.py
MOD = 10**9 + 7
N, K = map(int, input().split())


def mod_pow(x, n):
    res = 1
    while n > 0:
        if n & 1:
            res = res * x % MOD
        x = x * x % MOD
        n >>= 1
    return res


def inv(x):
    return mod_pow(x, MOD - 2)


P = [1] * (N + 1)
Q = [1] * (N + 1)
for i in range(1, N + 1):
    P[i] = P[i - 1] * i % MOD
Q[N] = inv(P[N])
for i in range(N, 0, -1):
    Q[i - 1] = Q[i] * i % MOD


def mod_com(n, k):
    if k == 0:
        return 1
    if n < 0 or k < 0 or k > n:
        return 0
    return P[n] * Q[n - k] % MOD * Q[k] % MOD


for i in range(1, K + 1):
    res = mod_com(N - K + 1, i) * mod_com(K - 1, i - 1) % MOD
    print(res)
